packages/jaeger-ui/src/components/TracePage/TraceTimelineViewer/VirtualizedTraceView.tsx

```diff
@@ -159,6 +159,8 @@ function getCssClasses(currentViewRange: [number, number]) {
   });
 }
 
+const memoizedSpanByID = memoizeOne((spans: Span[]) => new Map(spans.map(x => [x.spanID, x])));
```

Comment:
fyi https://github.com/jaegertracing/jaeger-ui/issues/2179#issuecomment-2228691694

Reply:
I know, but I am not sure if there is a good place for this data.  

Creating a Map or a Set should not have measurable impact on performance, because it is usually done once per trace.  
Also, if any refactoring is needed, then it would probably be best to do this in another PR.